app/scripts/models/auth_brokers/base.js

```diff
@@ -175,6 +186,15 @@ define([
 
     _isForceAuthUrl: function () {
       return this.window.location.pathname === '/force_auth';
+    },
+
+    _isSignInUrl: function () {
+      var pathname = this.window.location.pathname;
+      return pathname === '/signin' || pathname === '/oauth/signin';
```

Comment:
Having the broker look into the path seems fragile and wrong to me. It may be the it's wrong to re-routing in the client. We could change the `GET /authorization` endpoint to handle this case and redirect us to the right endpoint to begin with (e.g., like it's doing with `action=signin` and `action=signup`).